keras/src/saving/file_editor.py

```diff
@@ -0,0 +1,185 @@
+import base64
+import io
+import json
+import zipfile
+
+import h5py
+
+try:
+    import IPython as ipython
+except ImportError:
```

Comment:
This would cause Keras to crash on import. Instead, you should do:

`ipython = None` and then raise the ImportError when you try to use it.